grace-core/src/Grace/Interpret.hs

```diff
@@ -46,8 +50,23 @@ import qualified System.FilePath      as FilePath
 data Input
     = Path FilePath
     -- ^ The path to the code
-    | Code Text
-    -- ^ Source code
+    | Code String Text
+    -- ^ Source code: @Code name content@
+
+-- | Settings for the interpreter
+newtype InterpretSettings = InterpretSettings
```

Comment:
Well, that's a good point. I will remove the `InterpreterSettings`. However, I'd like to keep the possibility to pass a callback function `URI -> m Text` that resolves the imports to `interpretWith`. The reason for this is that I'd like to move the builtin resolvers to either the `grace` package  or even a separate one, say `grace-resolver-default`, and do not deal with any kinds of URIs in the `grace-core` package at all.
The idea behind that is that URI imports are the one and only way to interface with the outside world during interpretation and `grace-core` provides an interface of that and nothing more.

Would that degree of customizability of the `interpretWith` function be something to be agreed on?

Reply:
From my point of view, the scope of the `grace` package is to separate out the command-line interface for people who don't need it, so I'd prefer the URI-related logic to still be in `grace-core` since I view that as being part of the core language independent of the command-line interface

Reply:
Ok, thank you!